en/guide/define_xml_element.md

```diff
@@ -0,0 +1,205 @@
+
+**Elements for a private project**
+
+- If your enums/messages won't ever sync back to the MAVLink project then define them wherever you like!
+
+
+## XML Message File Format
+
+The format used for defining enums and messages can be understood by reading [MAVLink XML Schema](../guide/xml_schema.md), and by inspecting [common.xml](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0/common.xml) and other dialect files. 
+
+> **Note** The format and structure of dialect files is formally defined in the XML Schema document: [mavschema.xsd](https://github.com/ArduPilot/pymavlink/blob/master/generator/mavschema.xsd).
+
+
+## Creating a Dialect File
+
+To create a new dialect file:
+1. Fork [mavlink/mavlink](https://github.com/mavlink/mavlink/) for your system and clone to your system
+1. Create a dialect file named after your MAVLink system (e.g. flight stack) in **message_definitions/v1.0/**
+1. Copy the following text into the new file.
+   ```xml
+   <?xml version="1.0"?>
+   <mavlink>
+
+       <include>common.xml</include>
+       <!-- <version>9</version> -->
+       <dialect>8</dialect>
+
+       <enums>
+           <!-- Enums are defined here (optional) -->
+       </enums>
+
+       <messages>
+           <!-- Messages are defined here (optional) -->
+       </messages>
+    
+   </mavlink>
+   ```
+   The template assumes that your dialect:
+   - imports **common.xml** (`<include>common.xml</include>`)
+   - takes its version from **common.xml**  (which is why the `version` tags are commented out).
+
+1. Update the includes:
+   - if the dialect is not based on **common.xml** remove the existing `include` line
+   - Add additional `<include> </include>` elements if you want to import additional files/dialects.
+     > **Note** Includes in nested files are ignored.
+1. Update the version:
+   - if the dialect is not based on **common.xml** uncomment the `<version>6</version>` line and use whatever version you like.
+       > **Note** The version specified in the current file is used by default. If it is not present then the version in included files is used. Most dialects inherit the version from **common.xml**.
+1. Update the `<dialect>8</dialect>` line to replace `8` with the next-largest unused dialect number (based on the other files in the folder).
+1. Optionally remove the `enums` or `messages` sections if you don't plan on declaring any of these elements.
+1. Add enums or messages as described in the following sections.
+1. Save the file, and create a PR to push it back to the **mavlink/mavlink** project repo.
+
+
+
+## Enums
+
+TBD
+
+<!-- 
+Assumptions
+- You can add new enums without risk provided they have a unique name
+- You can add new entries to an enum provided each has a unique name and value (what if clash)
+- You can add new entries for an existing enum in dialect, provided again that the value/name is unique
+- entry values and numbers must be unique ? - 
```

Comment:
Again, I'm not sure if pymavlink will test this. And there might be other generators that do not test it. So yes, please tell the users to take care.